clang/test/CodeGenSYCL/intel-fpga-local.cpp

```diff
@@ -1,49 +1,53 @@
-//CHECK: [[ANN9:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{max_private_copies:4}
-//CHECK: [[ANN16:@.str.[0-9]*]] = {{.*}}foobar
-//CHECK: [[ANN17:@.str.[0-9]*]] = {{.*}}{memory:MLAB}{sizeinfo:4,500}
-//CHECK: [[ANN18:@.str.[0-9]*]] = {{.*}}{memory:BLOCK_RAM}{sizeinfo:4,10,2}
-//CHECK: [[ANN19:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{numbanks:4}{bank_bits:4,5}
-//CHECK: [[ANN20:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4,10,2}{bankwidth:16}{numbanks:2}{bank_bits:0}
-//CHECK: [[ANN21:@.str.[0-9]*]] = {{.*}}{memory:MLAB}{sizeinfo:4}{numbanks:8}{bank_bits:5,4,3}
+//CHECK-DEVICE: [[ANN1:@.str]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{numbanks:4}
+//CHECK-DEVICE: [[ANN2:@.str.[0-9]*]] = {{.*}}{register:1}
+//CHECK-DEVICE: [[ANN3:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}
+//CHECK-DEVICE: [[ANN4:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{bankwidth:4}
+//CHECK-DEVICE: [[ANN5:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{max_private_copies:8}
+//CHECK-DEVICE: [[ANN10:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{pump:1}
+//CHECK-DEVICE: [[ANN11:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{pump:2}
+//CHECK-DEVICE: [[ANN12:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{merge:foo:depth}
+//CHECK-DEVICE: [[ANN13:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{merge:bar:width}
+//CHECK-DEVICE: [[ANN14:@.str.[0-9]*]] = {{.*}}{max_replicates:2}
+//CHECK-DEVICE: [[ANN15:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{simple_dual_port:1}
+//CHECK-DEVICE: [[ANN6:@.str.[0-9]*]] = {{.*}}{memory:BLOCK_RAM}{sizeinfo:4}
+//CHECK-DEVICE: [[ANN7:@.str.[0-9]*]] = {{.*}}{memory:MLAB}{sizeinfo:4}
+//CHECK-DEVICE: [[ANN8:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{bankwidth:8}
+//CHECK-DEVICE: [[ANN9:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{max_private_copies:4}
+//CHECK-BOTH: [[ANN16:@.str[.0-9]*]] = {{.*}}foobar
+//CHECK-DEVICE: [[ANN17:@.str.[0-9]*]] = {{.*}}{memory:MLAB}{sizeinfo:4,500}
+//CHECK-DEVICE: [[ANN18:@.str.[0-9]*]] = {{.*}}{memory:BLOCK_RAM}{sizeinfo:4,10,2}
+//CHECK-DEVICE: [[ANN19:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4}{numbanks:4}{bank_bits:4,5}
+//CHECK-DEVICE: [[ANN20:@.str.[0-9]*]] = {{.*}}{memory:DEFAULT}{sizeinfo:4,10,2}{bankwidth:16}{numbanks:2}{bank_bits:0}
+//CHECK-DEVICE: [[ANN21:@.str.[0-9]*]] = {{.*}}{memory:MLAB}{sizeinfo:4}{numbanks:8}{bank_bits:5,4,3}
 
-//CHECK: @llvm.global.annotations
-//CHECK-SAME: { i8* addrspacecast (i8 addrspace(1)* bitcast (i32 addrspace(1)* @_ZZ3quxiE5a_one to i8 addrspace(1)*) to i8*)
-//CHECK-SAME: [[ANN1]]{{.*}}i32 157
-//CHECK-SAME: { i8* addrspacecast (i8 addrspace(1)* bitcast (i32 addrspace(1)* @_ZZ3quxiE5b_two to i8 addrspace(1)*) to i8*)
-//CHECK-SAME: [[ANN16]]{{.*}}i32 161
+//CHECK-BOTH: @llvm.global.annotations
+//CHECK-DEVICE-SAME: { i8* addrspacecast (i8 addrspace(1)* bitcast (i32 addrspace(1)* @_ZZ3quxiE5a_one to i8 addrspace(1)*) to i8*)
```

Comment:
I used to think that the FileCheck runs for separate check-prefixes do not share the same `CHECK-` lines, and therefore the `CHECK-DEVICE-SAME` wouldn't "attach" to a `CHECK-BOTH`. Am I simply wrong here?

Reply:
I can answer this way - the test is correctly passed, so I do an assumption to think that `CHECK-DEVICE-SAME` is really "attached" to `CHECK-BOTH`.

Reply:
Yes, I've checked this via:
`FileCheck test.txt --input-file test.txt --check-prefixes=CHECK-ALL,CHECK-SPECIFIC
```
// CHECK-ALL: All // CHECK-SPECIFIC-SAME: Specific                                                                                                                                             AllSpecific
```
Cool to know!